File: src/main/java/com/sadetec/model/ProductCategory_.java
package com.sadetec.model;

import java.time.LocalDateTime;

import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;

@StaticMetamodel(ProductCategory.class)
public abstract class ProductCategory_ {

    // Raw attributes
    public static volatile SingularAttribute<ProductCategory, Integer> id;
    public static volatile SingularAttribute<ProductCategory, String> categoryName;
    public static volatile SingularAttribute<ProductCategory, String> categoryCode;
    public static volatile SingularAttribute<ProductCategory, String> categoryUrl;
    public static volatile SingularAttribute<ProductCategory, String> childCategoryName;
    public static volatile SingularAttribute<ProductCategory, String> childCategoryCode;
    public static volatile SingularAttribute<ProductCategory, String> childCategoryUrl;
    public static volatile SingularAttribute<ProductCategory, String> grandCategoryName;
    public static volatile SingularAttribute<ProductCategory, String> grandCategoryUrl;
    public static volatile SingularAttribute<ProductCategory, LocalDateTime> dateAdded;
}

File: src/main/resources/static/app/model/ProductCategory.js
Ext.define('tms.model.ProductCategory', {
    extend:'Ext.data.Model',
    proxy: tms.getAjaxProxy('productCategory'),
    fields:[
    		{name:'id', type:'string'}
    		,{name:'categoryName', type:'string'}
    		,{name:'categoryCode', type:'string'}
    		,{name:'categoryUrl', type:'string'}
    		,{name:'childCategoryName', type:'string'}
    		,{name:'childCategoryCode', type:'string'}
    		,{name:'childCategoryUrl', type:'string'}
    		,{name:'grandCategoryName', type:'string'}
    		,{name:'grandCategoryUrl', type:'string'}
    		,{name:'dateAdded', type:'date'}
        ],
    getRecordName: function () {
        return this.get("id");
    }
});